resources/aws/auto_scaling_group_test.go

```diff
@@ -0,0 +1,69 @@
+package aws
+
+import (
+	"testing"
+
+	"github.com/aws/aws-sdk-go/service/autoscaling"
+)
+
+func TestIsLegacyASG(t *testing.T) {
+	nameOk := "my-asg"
+	clusterOk := "my-cluster"
+	tagNameOK := tagKeyName
+	tagClusterOK := tagKeyCluster
+	tagCF := "aws:cloudformation:stack-id"
+
+	t.Run("name and cluster match, no CF tag", func(t *testing.T) {
```

Comment:
We usually write table driven tests. They are easier to extend, more DRY and more readable. Right now I have to closely look for the `!` before `IsLegacyASG` to see if the expected result is true or false. Would you mind rewriting these tests? Check any of our golang projects and you should find this kind of tests. See for instance https://github.com/giantswarm/ipam/blob/032da5280d2dff234739ac38c9445e4d9cfcf2cd/ipam_test.go#L35.

Reply:
Sure, i started with them but because of the tags the testCases were not that readable, will push the changes to use them

Reply:
Done, i've moved also the `Run` call description to the table, to make things more clear, pls take a look